pkg/cgroups/utils.go

```diff
@@ -173,3 +191,25 @@ func (c *CgroupControl) createCgroupDirectory(controller string) (bool, error) {
 	}
 	return true, nil
 }
+
+// checkAndRecreatePath is a helper function used to validate the existece of the parent cgroup dir
+func checkAndRecreatePath(root string) error {
+	current := "/sys/fs"
+	if strings.Contains(root, "/sys/fs") {
+		root = strings.Split(root, "/sys/fs")[1]
+	} else {
+		return fmt.Errorf("could not validate cgroup path, must be rooted at /sys/fs")
+	}
+	split := strings.Split(root, "/")
+	for _, dir := range split {
```

Comment:
Doesn't mkdirall handle this?

Or github/containers/storage/pkg/idtools

idtools.MkdirAllAs